### PR TITLE
eyewitness: init at 20230525.1

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Tochiaha** [comment]
@Aleksanaa can you review, need to import selenium but having challenges achieving it.

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  src = fetchFromGitHub {
+    owner = "redsiege";
+    repo = "EyeWitness";
+    rev = "v${version}";
+    hash = "sha256-nSPpPbwqagc5EadQ4AHgLhjQ0kDjmbdcwE/PL5FDL4I=";
+  };
+
+  buildsystems = with python311Packages; [ setuptools ];
+
+  dependencies = with python311Packages; [
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:24]:
```suggestion
  buildsystems = with python3Packages; [ setuptools ];

  dependencies = with python3Packages; [
```
Unless absolutely required, python3 should be used

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python311Packages
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:4]:
```suggestion
, python3Packages
```

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    firefox-esr
+    xvfb-run
+    geckodriver
+  ];
+
+  doCheck = false;
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/eyewitness}
+    #mkdir -p $out/share/eyewitness
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:41]:
```suggestion
```

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    cp -R * $out/share/eyewitness
+    cp -R ${python311Packages.selenium}/lib/python3.11/site-packages/selenium $out/share/eyewitness/Python/modules
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:43]:
```suggestion
    cp -R ${python3Packages.selenium}/lib/python3.11/site-packages/selenium $out/share/eyewitness/Python/modules
```
Symlinking is not possible here?

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    sed -i "/from ssl import SSLError as sslerr/a \sys.path.insert(1, '$out/share/eyewitness/Python/modules')" $out/share/eyewitness/Python/modules/selenium_module.py
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:44]: We should probably use patch here to be save

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    bin=$out/bin/eyewitness
+    cat > $bin << EOF
+    #!/usr/bin/env sh
+    exec python3.11 $out/share/eyewitness/Python/EyeWitness.py
+    EOF
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:49]: Why not use makeWrapper?

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    chmod u+x $bin
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:50]:
```suggestion
    chmod +x $bin
```

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  installPhase = ''
+    mkdir -p $out/{bin,share/eyewitness}
+    #mkdir -p $out/share/eyewitness
+    cp -R * $out/share/eyewitness
+    cp -R ${python311Packages.selenium}/lib/python3.11/site-packages/selenium $out/share/eyewitness/Python/modules
+    sed -i "/from ssl import SSLError as sslerr/a \sys.path.insert(1, '$out/share/eyewitness/Python/modules')" $out/share/eyewitness/Python/modules/selenium_module.py
+    bin=$out/bin/eyewitness
+    cat > $bin << EOF
+    #!/usr/bin/env sh
+    exec python3.11 $out/share/eyewitness/Python/EyeWitness.py
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:48]:
```suggestion
    exec ${python3.interpreter} $out/share/eyewitness/Python/EyeWitness.py
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash the commits

File: pkgs/by-name/ey/eyewitness/package.nix (hunk 6 above)
Comment (Tochiaha) [reply pkgs/by-name/ey/eyewitness/package.nix]: All problems solved now can you check for any more adjustments?

**Comment by Tochiaha** [comment]
#81418

**Comment by Tochiaha** [comment]
@SuperSandro2000

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  ] ++ [
+    firefox-esr
+    xvfb-run
+    geckodriver
+  ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,share/eyewitness}
+    cp -R * $out/share/eyewitness
+  '';
```
Comment (wegank) [inline pkgs/by-name/ey/eyewitness/package.nix:45]:
```suggestion
  installPhase = ''
    runHook preInstall

    mkdir -p $out/{bin,share/eyewitness}
    cp -R * $out/share/eyewitness

    runHook postInstall
  '';
```

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+  fixupPhase = ''
+    makeWrapper "${python3Packages.python.interpreter}" "$out/bin/eyewitness" \
+      --set PYTHONPATH "$PYTHONPATH" \
+      --add-flags "$out/share/eyewitness/Python/EyeWitness.py"
+  '';
```
Comment (wegank) [inline pkgs/by-name/ey/eyewitness/package.nix:51]:
```suggestion
  fixupPhase = ''
    runHook preFixup

    makeWrapper "${python3Packages.python.interpreter}" "$out/bin/eyewitness" \
      --set PYTHONPATH "$PYTHONPATH" \
      --add-flags "$out/share/eyewitness/Python/EyeWitness.py"

    runHook postFixup
  '';
```

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+  meta = with lib; {
+    description = "EyeWitness is designed to take screenshots of websites, provide some server header info, and identify default credentials if possible";
```
Comment (wegank) [inline pkgs/by-name/ey/eyewitness/package.nix:54]:
```suggestion
    description = "Take screenshots of websites, provide some server header info, and identify default credentials if possible";
```

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description

Please also shorten the description if possible.

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    selenium
+    fuzzywuzzy
+    pyvirtualdisplay
+    pylev
+    netaddr
+    pydevtool
+  ] ++ [
+    firefox-esr
+    xvfb-run
+    geckodriver
```
Comment (wegank) [inline pkgs/by-name/ey/eyewitness/package.nix:39]: Just out of curiosity, what are they for?

Comment (Tochiaha) [reply pkgs/by-name/ey/eyewitness/package.nix:39]: due to the nature of the script, selenium based projects require web browser engine i.e Gecko in this situation the program collects content like HTML, XML, images and GeckoDriver is a connecting link to the Firefox browser for Selenium based projects. It is a proxy that helps you communicate with Gecko-based browsers.

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+, fetchFromGitHub
+, python3Packages
+, cmake
+, xvfb-run
+, firefox-esr
+, geckodriver
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
```
Comment (Aleksanaa) [inline pkgs/by-name/ey/eyewitness/package.nix:12]: Please use python3Packages.buildPythonApplication. See the nixpkgs manual.

Comment (Tochiaha) [reply pkgs/by-name/ey/eyewitness/package.nix]: specified pyproject = false;

**Comment by Tochiaha** [comment]
> Please squash the commits

Commits squashed.

File: pkgs/by-name/ey/eyewitness/package.nix [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+, xvfb-run
+, firefox-esr
+, geckodriver
+, makeWrapper
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "eye-witness";
+  version = "20230525.1";
+  pyproject = false;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ey/eyewitness/package.nix:14]:
```suggestion
  format = "other";
```

Comment (Tochiaha) [reply pkgs/by-name/ey/eyewitness/package.nix]: Done. Thank you!

**Comment by Tochiaha** [comment]
> Please squash the commits

Commits squashed.

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**